src/btree/row_srch.c

```diff
@@ -102,7 +102,18 @@ __wt_search_insert(
     match = skiphigh = skiplow = 0;
     ins = last_ins = NULL;
     for (i = WT_SKIP_MAXDEPTH - 1, insp = &ins_head->head[i]; i >= 0;) {
-        if ((ins = *insp) == NULL) {
+        /*
+         * The algorithm requires that the skip list insert pointer is only read once within the
+         * loop. Whilst the compiler can change the code in a way that it reads the insert pointer
```

Comment:
Extremely minor nit: I would just say "While" instead of using the more archaic "Whilst".

Reply:
@keitharnoldsmith , I actually requested this to read either "Whilst" or "Whereas," or just to join the two sentences, but this was my mistake. As these are subordinate conjunctions, it can be misleading to have only one clause in the sentence, especially if we start with the sentence with them. I think the better solution here would have been to just join the two sentences. That was my mistake, sorry @quchenhao!